spec/datadog/core/environment/execution_spec.rb

```diff
@@ -114,5 +118,94 @@
         end
       end
     end
+
+    context 'when webmock has enabled net-http adapter' do
+      before do
+        allow(described_class).to receive(:repl?).and_return(false)
+        allow(described_class).to receive(:test?).and_return(false)
+        allow(described_class).to receive(:rails_development?).and_return(false)
+
+        WebMock.enable!
+      end
+
+      it { is_expected.to eq(true) }
+    end
+  end
+
+  describe '.webmock_enabled?' do
+    context 'when missing constant `WebMock`' do
+      it do
+        hide_const('::WebMock')
+        expect(described_class).not_to be_webmock_enabled
+      end
+    end
+
+    context 'when missing constant `Net::HTTP`' do
+      it do
+        hide_const('::Net::HTTP')
+        expect(described_class).not_to be_webmock_enabled
+      end
+    end
+
+    context 'when `WebMock` and `Net::HTTP` constants both exist' do
+      before do
+        WebMock.enable!
```

Comment:
could it make sense to disable WebMock in `after` block? wouldn't it spill in the global scope for another tests?